Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -219,8 +242,72 @@ public class PBXProjGenerator {
             pbxProject.projects = subprojects
         }
 
-        try project.targets.forEach(generateTarget)
-        try project.aggregateTargets.forEach(generateAggregateTarget)
+        let generateTargetsGroup = DispatchGroup()
+        var targetErrors: [Error] = []
+        var aggregateTargetsErrors: [Error] = []
+
+        var generatedTargetNames: [String] = []
+        var targetsPendingDependencyGeneration: [Target] = project.targets
+        while generatedTargetNames.count != project.targets.count {
+            var targetsReadyToGenerate = targetsPendingDependencyGeneration.filter({ [project] target in
+                let dependencies = target.getAllDependencies(usingProject: project)
+                return dependencies.reduce(into: true) { (result: inout Bool, dependency) in
+                    if dependency.reference.contains("/") {
```

Comment:
I am not sure about this assumption, please help review, we can probably get rid of this if check as well.

But this helps me parallelise a few targets which just depend on an existing Xcode project reference